### DISCUSSION TITLE
run different query based on value of select combobox

### DISCUSSION THREAD

**Discussion by Lan-Phan** [illacloud/illa-builder#1027] [General]
Hello,

I have 1 simple page below:
+ 3 different actions named: daily_insight, weekly_insight and monthly_insight
+ Combo box with 3 options: Daily (value = daily_insight), Weekly (value = weekly_insight) and Monthly (value = monthly_insight)
+ 1 table to display result from 1 of 3 actions above

Is it possible for me to design app in which user select:
+ Daily -> daily_insight runs and load result into table
+ Weekly -> weekly_insight runs and load result into table
+ Monthly -> monthly_insight runs and load result into table

**Comment by Lan-Phan** [comment]
I tries to use event handler of combobox, but seems it cannot fulfill my demand:
+ use event Change -> action Control component. I think I should change data source of table to value of combobox, but it's pity that list of components doesn't include my table :(
+ use event Change -> action Trigger action, which I think I should trigger running different queries but the list of queries is static and no option to select query based on value of combobox

**Comment by AruSeito** [comment]
You can use action with 'Run action automatically when inputs change' Mod
Let's take MySQL as an example
1. drag and drop the Select widget, and change the options with what you need.
2. create an action with MySql
3. input SQL: select * from users where id > {{select1.value}}
4. change the action mod to Run action automatically when inputs change
5. drag and drop the Table widget, and change the data sources to mysql1
6. Then you can select something







example: select * from users where id > {{select1.value}}

**Comment by Lan-Phan** [comment]
Hi @AruSeito 

Your example is for 1 action, what I need is 3 different actions because they're completely different.
How can I achieve it?

**Reply by AruSeito** [reply]
Hello, you can create an action with what type of transform. but transform has a bug! I will fix it soon
![image](https://user-images.githubusercontent.com/41471697/222315577-058f3774-eac2-4320-8d61-ec191bde941d.png)
you can write 
```js
switch (select1.value){
  case "a":
    return action1.data
  case "b":
    return action2.data
  case "b":
    return action3.data
}
```

**Comment by Lan-Phan** [comment]
Hi @AruSeito ,

Have you fixed bug of transformer yet? I'm looking forward to use this function ...

Thanks.

**Reply by AruSeito** [reply]
We apologize that the issue has not been fixed yet. Currently, we are actively developing a new version, and we do not have enough time and staff to address this problem. We anticipate that we will not be able to resolve it until after the new version is released.

**Comment by Lan-Phan** [comment]
Hi @AruSeito ,

I see that Illa cloud have some new versions lately, I wonder these new versions include patch for this transformer bug or not? Please let I know, thanks so much.

**Reply by AruSeito** [reply]
@Lan-Phan Sorry, there are currently no fixes available. We will release a fixed version tomorrow evening(UTC).

**Reply by AruSeito** [reply]
Hello,@Lan-Phan 
We just released a fixed version, you can try again
